File: eTierMeeting_v2/SPA/src/app/_core/_models/production/T5/efficiencyExtKanbanModel.ts
import { FactoryInfo } from "./efficiency";

export interface EfficiencyExtKanbanModel {
  title: string;
  showValues: number;
  chartUnit: string;
  isActive: boolean;
  digits: number;
  labels: string[];
  data: FactoryDataChartExt[];
}

export interface FactoryDataChartExt {
  name: string;
  value: (number | null)[];
  actualQty: (number | null)[];
}

export interface FactoryExtInfo {
  id: string;
  name: string;
}

export interface MonthOfYearExt {
  month: number;
  year: number;
}

export interface EffiencyExtKanbanParam {
  type: string;
  factorys: FactoryInfo[];
  months: MonthOfYearExt[];
  years: number[];
  weeks: WeekExt[];
  seasons: SeasonExt[];
}

export interface WeekExt {
  name: string;
  weekStart: string;
  weekFinish: string;
}

export interface SeasonExt {
  name: string;
  seasonStart: string;
  seasonFinish: string;
}

File: eTierMeeting_v2/API/DTO/Production/T5/EfficiencyKanban/EfficiencyExtKanbanModel.cs
using System.Collections.Generic;

namespace eTierV2_API.DTO.Production.T5.EfficiencyKanban
{
    public class EfficiencyExtKanbanModel
    {
        public string Title { get; set; }
        public int ShowValues { get; set; }
        public string ChartUnit { get; set; }
        public bool IsActive { get; set; }
        public int Digits { get; set; }
        public List<string> Labels { get; set; }
        public List<FactoryDataChartExt> Data { get; set; }
    }

    public class FactoryDataChartExt
    {
        public string Name { get; set; }
        public List<decimal?> Value { get; set; }
        public List<int?> ActualQty { get; set; }
    }
}